clients/src/main/java/org/apache/kafka/clients/tools/VerifiableProducer.java

```diff
@@ -124,6 +132,23 @@ private static ArgumentParser argParser() {
                 .choices(0, 1, -1)
                 .metavar("ACKS")
                 .help("Acks required on each produced message. See Kafka docs on request.required.acks for details.");
+        
+        parser.addArgument("--producer-prop")
+                .action(Arguments.append())
+                .required(false)
+                .type(String.class)
+                .dest("userSpecifiedProducerProps")
+                .metavar("PRODUCER-PROP")
+                .help("Any custom producer properties of the form key=val.");
+
+        parser.addArgument("--close-timeout")
+                .action(store())
+                .required(false)
+                .setDefault(10L)
```

Comment:
How much more backlogged can it get than the buffer size? Aren't those callbacks invoked in the producer's network thread such that they would block any further sends until they complete? 

In any case, this probably means we need to reduce the cost of this logging. It probably needs some profiling to determine what the issue is for sure, but some obvious things that might be affecting how expensive this is:

* `System.out` will flush with every newline, i.e. for every call to println. This could add up since each line isn't that long.
* There's a lot of redundant work being done, e.g. creating JSONObject, getting the class name, etc.
* How fast is this JSON library? It could end up being faster to just do the encoding manually even though that's less than ideal.
* We're converting to a string, then writing to `System.out` rather than writing directly.

Also, be careful evaluating this by running directly in the console. If the issue has anything to do with the output, involving a console can significantly affect performance. You're better off logging something so you can tell when Ctrl-C was hit, then using the subsequent log entries and their timestamps to see how many more events there were and how long they took. Along the same lines, is this log being written locally and then synced via ducktape or is it running via SSH and with the SSH output being processed on the test runner? The SSH connection could also end up being a bottleneck.